src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -407,22 +400,13 @@ public static <K, V extends TopLevelItem> Map<K, V> loadChildren(AbstractFolder<
 
     @Override
     public String getItemName(File dir, I item) {
-        String name = childNameGenerator().readItemName(dir);
-        String computedName = childNameGenerator().itemNameFromItem(this, item);
-        if (computedName == null) {
-            computedName = dir.getName();
-        }
-        if (!computedName.equals(name)) {
-            throw new IllegalStateException("Computed name '" + computedName + "' does not match name read from file '" + name + "'");
+        String name = childNameGenerator().itemNameFromItem(this, item);
+        if (name == null) {
+            name = dir.getName();
         }
         return name;
     }
 
-    protected final I itemsPut(String name, I item) {
```

Comment:
This method seems to be used by a CloudBees proprietary plugin.

Reply:
Restored with a deprecation warning asking people to inline this (now) trivial method.